mne/io/eyelink/_utils.py

```diff
@@ -213,7 +156,8 @@ def _find_overlaps(df, max_time=0.05):
     """
     pd = _check_pandas_installed()
 
-    df = df.copy()
+    if not len(df):
+        return
```

Comment:
No worries!

I made sure to double check, I don't think I ever needed the copy in the first place: Even when we had the copy, we were re-assigning the new dataframe to itself anyways (currently in lines 250-256 of mne.io.eyelink.eyelink.py):

```
 if find_overlaps:
            for key in self.dataframes:
                if key not in ["blinks", "fixations", "saccades"]:
                    continue
                self.dataframes[key] = _find_overlaps(
                    self.dataframes[key], max_time=overlap_threshold
                )
```

Modifying the dataframe is intended, because the ocular `mne.Annotations` are made from these dataframes. For example, After `_find_overlaps`, a dataframe that had rows for blinks in both the left and right eyes, now has a single row for each blink, representing the blink in both eyes (assuming that the separate blinks were less than 50ms apart) - and a single annotation would be created for this blink.

Does that sound reasonable to you?

FWIW - The reason I deleted the `df.copy`, is I was trying to be more conservative with resources. The problematic file that led to this bug report was a 200mb text file with 4 million lines, so I didn't want to do an unnecessary copy.

Reply:
yes if you're overwriting later anyway then no need to copy. Sorry I didn't notice the overwriting.

Reply:
No worries!